cmake/FindFirebaseCore.cmake

```diff
@@ -0,0 +1,57 @@
+# Copyright 2017 Google
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+find_library(
+  FIREBASECORE_LIBRARY
+  FirebaseCore
+  PATHS ${FIREBASE_BINARY_DIR}/Frameworks
+)
+
+include(FindPackageHandleStandardArgs)
+find_package_handle_standard_args(
+  FirebaseCore
+  DEFAULT_MSG
+  FIREBASECORE_LIBRARY
+)
+
+if(FIREBASECORE_FOUND)
+  # Emulate Cocoapods behavior which makes all headers available unqualified.
+  set(
+    FIREBASECORE_INCLUDE_DIRS
+    ${FIREBASECORE_LIBRARY}/Headers
+    ${FIREBASECORE_LIBRARY}/PrivateHeaders
+  )
+
+  # TODO(mcg): on iOS this should depend on UIKit.
```

Comment:
Why not Foundation for both iOS and macOS?

Reply:
Foundation is insufficient: [FIRAppEnvironmentUtil](https://github.com/firebase/firebase-ios-sdk/blob/f5cabf4feb54fe39940873fe9a7de8c98c4d1bb2/Firebase/Core/third_party/FIRAppEnvironmentUtil.h#L49) depends upon `NSApplication` from AppKit or `UIApplication` from UIKit conditionally.